### ISSUE TITLE
Hard/Soft Line Wrap after max characters works incorrectly

### ISSUE THREAD

**Issue by ghostwheel** [texstudio-org/texstudio#707]
Maximal characters are somehow used incorrectly. I'm using the option (in Adv.Editor) Line wrapping: hard line wrap after max characters. (or soft, it doesn't matter)
Somehow the max characters setting has to be halved to work as needed.
My Editor font is Courier New 14.
If I set max characters to 80, which is what I want, then the wrapping happens at around column 146. I also don't see the hard line on screen.
If I set max characters to 45, wrapping happens at 80, and I see the line on screen around where it should be.
But, when TeXstudio starts, if max characters is set to 45, then the lines wrap at 45, the break line shows at around 80. Then I have to change the max characters to 80, and back to 45. Then everything is where I want it.
If I start TeXstudio with max characters set to 80, then lines are wrapped at 80, the break line is beyond the end, and typing new lines will break them at ~146.

### Environment
- TeXstudio: 2.12.16
- Qt: 5.12.1 R
- OS:Windows 10, 1903
- TeX distribution: MiKTeX 2.9.7000 64-bit

### Expected behavior

When setting maximal characters to 80 in settings->Adv Editor, and line wrapping "soft line wrap after max. characters" or "hard line wrap after max. characters", line should wrap after 80 characters.

### Actual behavior

When setting to 80, line wraps at around 146, and right indicator for end of line is at around 146.
![max is 80](https://user-images.githubusercontent.com/235726/63893487-83d4e600-c9a7-11e9-9352-fc981bd961f7.PNG)
![max is 40](https://user-images.githubusercontent.com/235726/63893506-94855c00-c9a7-11e9-876f-2a089e83543a.PNG)
. When setting to 80, quitting and restarting texstudio, text wraps at 80, but end of line indicator is at 146.
![max is 80 after restart](https://user-images.githubusercontent.com/235726/63893516-9bac6a00-c9a7-11e9-9e91-a5f97c2f3641.PNG)
![max is 40 after restart](https://user-images.githubusercontent.com/235726/63893523-a109b480-c9a7-11e9-82ed-77b616399c48.PNG)

### How to reproduce
In Settings->Adv Editor set max characters to 80, and line wrapping "soft line wrap after max. characters". Enter in text editor a long line.

**Comment by timhoffm** [comment]
Are you using a high-resolution screen with scaling in windows? This sounds much like a scaling problem.

**Comment by timhoffm** [comment]
Do you have different magnification factors on the tablet vs. external monitor. IIRC we (or Qt? - depending on the Qt version) do not handle magnifications screen-dependent.

**Comment by ghostwheel** [comment]
Yes, you are right! When the scaling is different on the two displays I have the problems, and when it is the same, not. The problem is that the DPI is very different between the tablet and the screen. But I can work with it now. 

It is still strange that the only place I see the problem is in text wrapping.

**Comment by homocomputeris** [comment]
I have a similar problem in
```
TeXstudio 3.0.1 (git n/a)
Using Qt Version 5.15.1, compiled with Qt 5.15.1 R
```
on a 3840×2160 running in Gnome Shell 3.38.1 with the scale factor 2.
If I set wrap at 80, it wraps at 90-sh.

**Comment by sunderme** [comment]
please be aware that your font is probably *not* monospaced, i.e. the width of the characters varies depending on the character. txs uses average with as line width and breaks when that width in pixel is reached ...

**Comment by homocomputeris** [comment]
I use Noto Sans Mono

**Comment by muzimuzhi** [comment]
@homocomputeris Perhaps you can open a new issue, with a screenshot showing the wrapping problem.
